src/Compilers/VisualBasic/Portable/Lowering/Instrumentation/DynamicAnalysisInjector.vb

```diff
@@ -0,0 +1,300 @@
+            Dim spanPosition As FileLinePositionSpan = location.GetMappedLineSpan()
+            Dim path As String = spanPosition.Path
+            If path.Length = 0 Then
+                path = syntaxForSpan.SyntaxTree.FilePath
+            End If
+
+            Dim spansIndex As Integer = _spansBuilder.Count
+            _spansBuilder.Add(New SourceSpan(_debugDocumentProvider.Invoke(path, ""), spanPosition.StartLinePosition.Line, spanPosition.StartLinePosition.Character, spanPosition.EndLinePosition.Line, spanPosition.EndLinePosition.Character))
+
+            ' Generate "_payload(pointIndex) = True".
+
+            Dim payloadCell As BoundArrayAccess = statementFactory.ArrayAccess(statementFactory.Local(_methodPayload, False), True, statementFactory.Literal(spansIndex))
+            Return statementFactory.Assignment(payloadCell, statementFactory.Literal(True))
+        End Function
+
+        Private Shared Function SyntaxForSpan(statement As BoundStatement) As VisualBasicSyntaxNode
+            SyntaxForSpan = statement.Syntax
+
+            Select Case statement.Kind
+                Case BoundKind.IfStatement
+                    SyntaxForSpan = DirectCast(statement, BoundIfStatement).Condition.Syntax
+                Case BoundKind.WhileStatement
+                    SyntaxForSpan = DirectCast(statement, BoundWhileStatement).Condition.Syntax
+                Case BoundKind.ForEachStatement
+                    SyntaxForSpan = DirectCast(statement, BoundForEachStatement).Collection.Syntax
+                Case BoundKind.DoLoopStatement
+                    Dim condition As BoundExpression = DirectCast(statement, BoundDoLoopStatement).ConditionOpt
+                    If condition IsNot Nothing Then
+                        SyntaxForSpan = condition.Syntax
+                    End If
+                Case BoundKind.UsingStatement
+                    Dim usingStatement As BoundUsingStatement = DirectCast(statement, BoundUsingStatement)
+                    SyntaxForSpan = If(usingStatement.ResourceExpressionOpt IsNot Nothing, DirectCast(usingStatement.ResourceExpressionOpt, BoundNode), usingStatement).Syntax
+                Case BoundKind.SyncLockStatement
+                    SyntaxForSpan = DirectCast(statement, BoundSyncLockStatement).LockExpression.Syntax
+                Case BoundKind.SelectStatement
+                    SyntaxForSpan = DirectCast(statement, BoundSelectStatement).ExpressionStatement.Expression.Syntax
+                Case Else
+                    SyntaxForSpan = statement.Syntax
+            End Select
+        End Function
+
+        Private Shared Function MethodHasExplicitBlock(method As MethodSymbol) As Boolean
+            Dim asSourceMethod As SourceMethodSymbol = TryCast(method.OriginalDefinition, SourceMethodSymbol)
+            If DirectCast(asSourceMethod, Object) IsNot Nothing Then
+                Return TypeOf asSourceMethod.Syntax Is MethodBlockSyntax
```

Comment:
We probably want to check for MethodBlockBaseSyntax here. Otherwise, constructors, operators and accessors are skipped.